recipe_scrapers/_schemaorg.py

```diff
@@ -79,31 +79,32 @@ def __init__(self, page_data):
                         self.ratingsdata[rating_id] = rating
 
         for syntax in SYNTAXES:
-            # Make sure entries of type Recipe are always parsed first
-            syntax_data = data.get(syntax, [])
-            try:
-                index = [x.get("@type", "") for x in syntax_data].index("Recipe")
-                syntax_data.insert(0, syntax_data.pop(index))
-            except ValueError:
-                pass
-
-            for item in syntax_data:
+            for item in data.get(syntax, []):
                 if SCHEMA_ORG_HOST not in item.get("@context", ""):
                     continue
 
                 # If the item itself is a recipe, then use it directly as our datasource
                 if recipe := self._find_entity(item, "Recipe"):
-                    self.format = syntax
-                    self.data = recipe
-                    return
-
+                    pass
                 # If the item is a webpage and describes a recipe entity, use the entity as our datasource
-                if self._contains_schematype(item, "WebPage"):
-                    main_entity = item.get("mainEntity", {})
-                    if self._contains_schematype(main_entity, "Recipe"):
+                elif self._contains_schematype(item, "WebPage") and (
+                    recipe := item.get("mainEntity", {})
+                ):
+                    pass
+                else:
+                    continue
+                if not self._contains_schematype(recipe, "Recipe"):
+                    continue
+
+                self.data = self.data or recipe
+                for prop in ("@id", "name"):
+                    existing_value = self.data.get(prop)
+                    encountered_value = recipe.get(prop)
+                    if existing_value and encountered_value == existing_value:
                         self.format = syntax
```

Comment:
Todo: we're potentially allowing mixed-syntax format, with these changes.  It would be possible for _both_ `ld+json` (`schema.org` in `<script>` tags) and HTML attribute (`itemtype`, `itemprop`, etc) schema mappings to be combined into the results.

Reply:
This is an important note. In the longer term, I think we should start using the opengraph syntax parsing from the `extruct` package itself, and we should keep things like this in mind.
Could you add this as a TODO comment in the code? I think it's important.

Reply:
There's a terminology detail to figure out too I think.  I think we support two metadata _standards_ (`schema.org`, OpenGraph), and within `schema.org` we support two.. formats?  or syntaxes?  (`microdata`, `ld+json`).  Does OpenGraph also allow multiple formats?  (maybe a question for a separate issue/time)